security/advancedtls/advancedtls.go

```diff
@@ -113,15 +137,32 @@ type ServerOptions struct {
 	// function every time asked to present certificates to the client when a new connection is
 	// established. This is known as peer certificate reloading.
 	GetCertificate func(*tls.ClientHelloInfo) (*tls.Certificate, error)
+	// VerifyPeer is a custom verification check after certificate signature check.
+	// If this is set, we will perform this customized check after doing the normal check(s)
+	// indicated by setting VType.
+	VerifyPeer CustomVerificationFunc
+	// RootCertificateOptions is only required when mutual TLS is enabled(RequireClientCert is true).
 	RootCertificateOptions
 	// If the server want the client to send certificates.
 	RequireClientCert bool
+	// Note that this field will only take into effect when we set RequireClientCert to true,
+	// meaning we are using mTLS and want to verify the client certificates.
+	// If setting this field to CertAndHostVerification, we would only perform the certificate
+	// verification, which has the same effect as setting to CertVerification, because we don't have
+	// hostname check on server side.
+	// If setting to CertVerification, we would perform verification for certificates sent from the
+	// client. Setting this field without proper custom authorization would leave the application
+	// susceptible to the MITM attack.
+	// If setting to SkipVerification, we would skip all the verification. Setting this field without
+	// proper custom authorization would leave the peer completely unauthenticated, and is
+	// highly discouraged unless you really have reasons to do so.
```

Comment:
We seem to be failing the call to create a new client credential if no custom verifyFunc is set and `VerificationType` is set to `SkipVerification`. But we just say that this is highly discouraged here. The comment should reflect behavior.